apps/frontend/src/app/Database/DataManager.ts

```diff
@@ -73,14 +73,36 @@ export class DataManager<CacheKey extends string, GOKey extends string, CacheVal
     this.listeners[key]?.forEach(cb => cb(key, reason, object))
     this.anyListeners.forEach(cb => cb(key, reason, object))
   }
-  remove(key: CacheKey) {
+  remove(key: CacheKey, notify = true) {
     const rem = this.data[key]
     delete this.data[key]
     this.removeStorageEntry(key)
 
-    this.trigger(key, "remove", rem)
+    if (notify) this.trigger(key, "remove", rem)
     delete this.listeners[key]
   }
+  swapId(oldKey: CacheKey, newKey: CacheKey, notify = false): boolean {
```

Comment:
I know it is pretty simple logic, but could you write a comment for when the swap/change should fail/succeed? maybe my brain is fried but reading comment for 2s > reading code for 30s